### ISSUE TITLE
[Feature] Clone all necessary repos before compilation

### ISSUE THREAD

**Issue by jacobleeharris** [ps2dev/ps2dev#97]
I believe the PS2DEV repo should clone all necessary repos before attempting to compile the toolchain.

On shoddy internet like my own, it is very annoying to have to restart the entire process each time my internet goes out when downloading a repository. If the repositories were all cloned before the compilation process, it would give me and others a much easier time compiling the toolchain without having to repeat the compilation process each time a repository fails to download at runtime. Alternatively, the scripts could be less error prone and allow repeating the last script ran before bailing out of the process entirely.

**Comment by jendo42** [comment]
I have had something in my mind using git submodules. But of how things are done now that would require rework of whole build system.

**Comment by jacobleeharris** [comment]
That would also work, though it would make switching dependencies out a tiny bit harder IMO.

**Comment by jendo42** [comment]
Yeah in this case is also fine to have some sort of config of revisions and the script fetching all the deps. But I think that the most annoying thing is the "recursive" calling the build scripts and the build scripts call another build scripts. And when you want to run it again the damn build script removes it all and builds it entirely again.
